Dockerfile.nightly fails to build on default branch ("Missing dependency: sail_lean_backend")

To reproduce:
- clone the repo
- run `docker build . -t sail -f .\Dockerfile.nightly --no-cache --progress plain`

The build fails and you get the [following output](https://pastebin.com/raw/FgFw2DAe). The error message says:
```
#10 [6/6] RUN opam pin -y add sail $PWD
#10 sha256:6033cfa6ee6ee07b152d74f65185f6d6dc8699e564d21c5bbcece04322d54404
#10 0.494 [WARNING] Running as root is not recommended
#10 2.379 [sail.0.18] synchronised (file:///sail-src)
#10 2.381 sail is now pinned to file:///sail-src (version 0.18)
#10 2.381 
#10 10.65 [ERROR] Package conflict!
#10 10.66   * Missing dependency:
#10 10.66     - sail_lean_backend
#10 10.66     unknown package
#10 10.66 
#10 10.66 [NOTE] Pinning command successful, but your installed packages may be out of sync.
#10 ERROR: executor failed running [/bin/sh -c opam pin -y add sail $PWD]: exit code: 20
```

Your on-push CI does not try to build the Dockerfile, so it is green.

--- a/.dockerignore
+++ b/.dockerignore
@@ -2,4 +2,5 @@
 *.md
 .gitignore
 .dockerignore
-Dockerfile
+Dockerfile*
+/.github

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -63,3 +63,13 @@ jobs:
       run: |
         eval $(opam env)
         etc/ci_core_tests.sh
+
+  build-docker:
+    runs-on: "ubuntu-latest"
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Build Dockerfile.nightly
+      run: |
+        docker build . -t sail -f ./Dockerfile.nightly --no-cache --progress plain

--- a/Dockerfile.nightly
+++ b/Dockerfile.nightly
@@ -11,6 +11,5 @@ WORKDIR /sail-src
 COPY . /sail-src
 
 # build
-RUN opam pin -y add sail $PWD
-
+RUN opam install --yes .
 ENTRYPOINT [ "/sail-src/docker_entry_point.sh" ]

